### PR TITLE
Remove access to `_short_id` in NeptuneLogger

### PR DESCRIPTION
**Pull request by Raalsky**
## What does this PR do?
Removes access to the internals of Run in NeptuneLogger (access to `run._short_id`)

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -266,6 +266,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Deprecated function `pytorch_lightning.callbacks.device_stats_monitor.prefix_metric_keys` ([#11254](https://github.com/PyTorchLightning/pytorch-lightning/pull/11254))
 
 
+- Remove access to `_short_id` in `NeptuneLogger` ([#11517](https://github.com/PyTorchLightning/pytorch-lightning/pull/11517))
```
Comment (rohitgr7) [inline CHANGELOG.md:269]:
```suggestion
- Removed access to `_short_id` in `NeptuneLogger` ([#11517](https://github.com/PyTorchLightning/pytorch-lightning/pull/11517))
```
also this is deprecated section. You might want to move to the `Removed` section down below.

**Review by rohitgr7 — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (Raalsky) [reply CHANGELOG.md]: Done. Thanks

**Review by Borda — APPROVED** [review]

File: pytorch_lightning/loggers/neptune.py [hunk 2]
```diff
@@ -293,9 +293,10 @@ def __init__(
     def _retrieve_run_data(self):
         try:
             self._run_instance.wait()
-            self._run_short_id = self.run._short_id  # skipcq: PYL-W0212
+            self._run_short_id = self._run_instance["sys/id"].fetch()
```
Comment (ananthsub) [inline pytorch_lightning/loggers/neptune.py:296]: Does this need a deprecation warning first before the breaking change is made?

Comment (Raalsky) [reply pytorch_lightning/loggers/neptune.py:296]: There is no breaking change. `_short_id` is internal property of our `Run` (which exists in the context of `neptune-client` library and `neptune` package). `["sys/id"].fetch()` is just a gentle way of access same value. There is no change from the user perspective.

Comment (Raalsky) [reply pytorch_lightning/loggers/neptune.py:296]: Or I'm missing something :wink:

**Review by tchaton — APPROVED** [review]
LGTM !

**Comment by Raalsky** [comment]
@awaelchli @carmocca @rohitgr7 @Borda @tchaton Merge? 🙏

**Comment by rohitgr7** [comment]
@Raalsky 

tests are running, it will auto merge now if they pass :)

**Pull request merged**